code/controllers/subsystem/movement/movement_types.dm

```diff
@@ -44,6 +44,12 @@
 	src.lifetime = timeout
 	return TRUE
 
+///proc that exists so we can check if this exact moveloop datum already exists (in terms of vars) and so we can stop it from needlessly create a new one to overwrite the old one
+/datum/move_loop/proc/compare_loops(datum/move_loop/loop_type, priority, flags, extra_info, delay = 1, timeout = INFINITY)
+	SHOULD_CALL_PARENT(TRUE)
+	if(loop_type == type && priority == src.priority && flags == src.flags && delay == src.delay && timeout == lifetime)
+		return TRUE
+
```

Comment:
Might be better to have a return FALSE here by default so you're not just relying on null being falsey

Reply:
in that case i guess i should do it for all the overwrites too trough